validation/api/api.lock.json

```diff
@@ -2,85 +2,83 @@
       "public func empty<T>() : [T]",
-      "public func enumerate<T>(array : [T]) : Types.Iter<(Nat, T)>",
-      "public func equal<T>(array1 : [T], array2 : [T], equal : (implicit : (T, T) -> Bool)) : Bool",
-      "public func filter<T>(array : [T], f : T -> Bool) : [T]",
-      "public func filterMap<T, R>(array : [T], f : T -> ?R) : [R]",
-      "public func find<T>(array : [T], predicate : T -> Bool) : ?T",
-      "public func findIndex<T>(array : [T], predicate : T -> Bool) : ?Nat",
-      "public func flatMap<T, R>(array : [T], k : T -> Types.Iter<R>) : [R]",
-      "public func flatten<T>(arrays : [[T]]) : [T]",
-      "public func foldLeft<T, A>(array : [T], base : A, combine : (A, T) -> A) : A",
-      "public func foldRight<T, A>(array : [T], base : A, combine : (T, A) -> A) : A",
-      "public func forEach<T>(array : [T], f : T -> ())",
+      "public func enumerate<T>(self : [T]) : Types.Iter<(Nat, T)>",
+      "public func equal<T>(self : [T], other : [T], equal : (implicit : (T, T) -> Bool)) : Bool",
+      "public func filter<T>(self : [T], f : T -> Bool) : [T]",
+      "public func filterMap<T, R>(self : [T], f : T -> ?R) : [R]",
+      "public func find<T>(self : [T], predicate : T -> Bool) : ?T",
+      "public func findIndex<T>(self : [T], predicate : T -> Bool) : ?Nat",
+      "public func flatMap<T, R>(self : [T], k : T -> Types.Iter<R>) : [R]",
+      "public func flatten<T>(self : [[T]]) : [T]",
+      "public func foldLeft<T, A>(self : [T], base : A, combine : (A, T) -> A) : A",
+      "public func foldRight<T, A>(self : [T], base : A, combine : (T, A) -> A) : A",
+      "public func forEach<T>(self : [T], f : T -> ())",
       "public func fromIter<T>(iter : Types.Iter<T>) : [T]",
       "public func fromVarArray<T>(varArray : [var T]) : [T]",
-      "public func indexOf<T>(array : [T], equal : (implicit : (T, T) -> Bool), element : T) : ?Nat",
-      "public func isEmpty<T>(array : [T]) : Bool",
-      "public func join<T>(arrays : Types.Iter<[T]>) : [T]",
-      "public func keys<T>(array : [T]) : Types.Iter<Nat>",
-      "public func lastIndexOf<T>(array : [T], equal : (implicit : (T, T) -> Bool), element : T) : ?Nat",
-      "public func map<T, R>(array : [T], f : T -> R) : [R]",
-      "public func mapEntries<T, R>(array : [T], f : (T, Nat) -> R) : [R]",
-      "public func mapResult<T, R, E>(array : [T], f : T -> Types.Result<R, E>) : Types.Result<[R], E>",
-      "public func nextIndexOf<T>(array : [T], equal : (implicit : (T, T) -> Bool), element : T, fromInclusive : Nat) : ?Nat",
-      "public func prevIndexOf<T>(array : [T], equal : (implicit : (T, T) -> Bool), element : T, fromExclusive : Nat) : ?Nat",
-      "public func range<T>(array : [T], fromInclusive : Int, toExclusive : Int) : Types.Iter<T>",
+      "public func indexOf<T>(self : [T], equal : (implicit : (T, T) -> Bool), element : T) : ?Nat",
+      "public func intoArray<C, T>(self : C, values : (implicit : C -> Types.Iter<T>)) : [T]",
```

Comment:
This is new

Reply:
Actually, I'm not sore how useful this is really. The implicit argument can't be generic and implicitly instantiated, so I don't think this will work with most data structures, right?
Maybe leave out for now?